### PR TITLE
Implement support for daita with multihop

### PR DESCRIPTION
**Pull request by Pururun**
- - -
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/7230)

### CODE REVIEW

**Review by Rawa — COMMENTED** [review]
> Reviewed 30 of 93 files at r1, 36 of 59 files at r2, 1 of 61 files at r4.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/7230)* status: 30 of 111 files reviewed, 2 unresolved discussions (waiting on @Pururun)

___
*[`android/lib/daemon-grpc/src/main/kotlin/net/mullvad/mullvadvpn/lib/daemon/grpc/ManagementService.kt` line 512 at r2](https://reviewable.io/reviews/mullvad/mullvadvpn-app/7230#-OCbbqg95xjonF68IsAK:-OCbbqg95xjonF68IsAL:b-ktkxgv) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/0e5b01bdd9991047df821f0ebc2385d31a5c8bc3/android/lib/daemon-grpc/src/main/kotlin/net/mullvad/mullvadvpn/lib/daemon/grpc/ManagementService.kt#L512)):*
> ```Kotlin
>             .mapEmpty()
> 
>     suspend fun setDaitaEnabled(enabled: Boolean): Either<SetDaitaSettingsError, Unit> =
> ```

Should we use this message instead?
```  rpc SetEnableDaita(google.protobuf.BoolValue) returns (google.protobuf.Empty) {}```
___
*[`android/lib/daemon-grpc/src/main/kotlin/net/mullvad/mullvadvpn/lib/daemon/grpc/ManagementService.kt` line 521 at r2](https://reviewable.io/reviews/mullvad/mullvadvpn-app/7230#-OCbby1hEBnunp_LKYC6:-OCbby1hEBnunp_LKYC7:bb5bp81) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/0e5b01bdd9991047df821f0ebc2385d31a5c8bc3/android/lib/daemon-grpc/src/main/kotlin/net/mullvad/mullvadvpn/lib/daemon/grpc/ManagementService.kt#L521)):*
> ```Kotlin
>             .mapEmpty()
> 
>     suspend fun setDaitaDirectOnly(enabled: Boolean): Either<SetDaitaSettingsError, Unit> =
> ```

Should we use this message instead?
```  rpc SetDaitaDirectOnly(google.protobuf.BoolValue) returns (google.protobuf.Empty) {}```

**Review by Pururun — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/7230)* status: 0 of 111 files reviewed, 2 unresolved discussions (waiting on @Rawa)

___
*[`android/lib/daemon-grpc/src/main/kotlin/net/mullvad/mullvadvpn/lib/daemon/grpc/ManagementService.kt` line 512 at r2](https://reviewable.io/reviews/mullvad/mullvadvpn-app/7230#-OCbbqg95xjonF68IsAK:-OCh8-VD1XcuMzQL0qSU:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/0e5b01bdd9991047df821f0ebc2385d31a5c8bc3/android/lib/daemon-grpc/src/main/kotlin/net/mullvad/mullvadvpn/lib/daemon/grpc/ManagementService.kt#L512)):*
<details><summary><i>Previously, Rawa (David Göransson) wrote…</i></summary><blockquote>

Should we use this message instead?
```  rpc SetEnableDaita(google.protobuf.BoolValue) returns (google.protobuf.Empty) {}```
</blockquote></details>

Done.
___
*[`android/lib/daemon-grpc/src/main/kotlin/net/mullvad/mullvadvpn/lib/daemon/grpc/ManagementService.kt` line 521 at r2](https://reviewable.io/reviews/mullvad/mullvadvpn-app/7230#-OCbby1hEBnunp_LKYC6:-OCh8-o7EYBfXpxFNNs-:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/0e5b01bdd9991047df821f0ebc2385d31a5c8bc3/android/lib/daemon-grpc/src/main/kotlin/net/mullvad/mullvadvpn/lib/daemon/grpc/ManagementService.kt#L521)):*
<details><summary><i>Previously, Rawa (David Göransson) wrote…</i></summary><blockquote>

Should we use this message instead?
```  rpc SetDaitaDirectOnly(google.protobuf.BoolValue) returns (google.protobuf.Empty) {}```
</blockquote></details>

Done.

File: android/app/src/main/kotlin/net/mullvad/mullvadvpn/compose/screen/DaitaScreen.kt [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    daitaConfirmationDialogResult.OnNavResultValue {
+        if (it) {
+            viewModel.setDirectOnly(true)
+        }
+    }
+
+    DaitaScreen(
+        state = state,
+        onDaitaEnabled = viewModel::setDaita,
+        onDirectOnly = { enable ->
```
Comment (Rawa) [inline android/app/src/main/kotlin/net/mullvad/mullvadvpn/compose/screen/DaitaScreen.kt:74]: onDirectOnlyClick?

File: android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/customlists/FilterCustomListsRelayItemUseCase.kt [hunk 2]
```diff
@@ -33,7 +34,7 @@ class FilterCustomListsRelayItemUseCase(
                 providers = selectedProviders,
                 daita =
                     shouldFilterByDaita(
-                        isDaitaEnabled = settings?.isDaitaEnabled() == true,
+                        isDaitaEnabled = settings?.daitaAndDirectOnly() == true,
```
Comment (Rawa) [inline android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/customlists/FilterCustomListsRelayItemUseCase.kt:37]: Should this be more explicit? `isDaitaEnabled` is seemingly from this logic specifically `isDirectOnlyDaitaEnabled`?

File: android/app/src/main/kotlin/net/mullvad/mullvadvpn/compose/screen/DaitaScreen.kt (hunk 1 above)
Comment (Pururun) [reply android/app/src/main/kotlin/net/mullvad/mullvadvpn/compose/screen/DaitaScreen.kt]: Fixed

File: android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/customlists/FilterCustomListsRelayItemUseCase.kt (hunk 2 above)
Comment (Pururun) [reply android/app/src/main/kotlin/net/mullvad/mullvadvpn/usecase/customlists/FilterCustomListsRelayItemUseCase.kt]: Changed

**Review by Rawa — DISMISSED** [review]

**Review by Rawa — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/7230#-:-OD5OBSEBNI3P-Gs5oT1:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

**Pull request merged**